### PR TITLE
State machine/game loop functionality 

### PR DESCRIPTION
**Pull request by konnorbeard**
In this commit: 
* redid the state machine/landing actions to help with feedback loop/input from user 
* fixed income tax
* adding housing cost to housingProperties
* added some helpers to player.js

@andrewjmeier @qstearns @JamesEVerhagen @kevinkim91 cr

### CODE REVIEW

File: src/board/board/space.js [hunk 1]
```diff
@@ -1,10 +1,16 @@
+require("../boardConstants");
+
 function Space(name) {
   this.name = name;
   this.occupier = null;
 }
 // every space needs a landing action
 Space.prototype.performLandingAction = function(game){
-  console.log(game.getCurrentPlayer().name + " landed on " + this.name);
+  return ["You landed on " + this.name + ". \n", POST_TURN];
```
Comment (andrewjmeier) [inline src/board/board/space.js]: any particular reason this went from current player's name to just you?

File: src/board/properties/housingProperty.js [hunk 2]
```diff
@@ -12,32 +12,26 @@ var PG_BLUE = 7;
...
 
 HousingProperty.prototype.performLandingAction = function(game) {
-
-    HousingProperty.super_.prototype.performLandingAction.call(this, game);
+  return HousingProperty.super_.prototype.performLandingAction.call(this, game);
 };
 
 HousingProperty.prototype.getRent = function(game) {
 
+  if (!this.owner) return 0;
```
Comment (andrewjmeier) [inline src/board/properties/housingProperty.js]: What's this?

File: src/board/properties/property.js [hunk 3]
```diff
@@ -12,29 +13,40 @@ function Property(name, cost, propertyGroup) {
...
+  } else if (player.owesRent(this)) { 
     var rent = this.getRent(game);
     player.payPlayer(rent, this.owner);
-  }
-  // todo  - finish hashing this out
-  Property.super_.prototype.performLandingAction.call(this, game);
+    actions[0] = actions[0].concat(" You payed $" + rent + " to " + this.owner.name + ". ");
+  } else if (!this.owner) { 
+    actions[0] = actions[0].concat(" It is unowned. ");
+    actions[1] = BUY_PROMPT;
```
Comment (andrewjmeier) [inline src/board/properties/property.js]: so what're the values in this array?

Comment (konnorbeard) [inline src/board/properties/property.js]: See comment in turn.js. [msg, nextstate]

File: src/game.js [hunk 4]
```diff
@@ -10,6 +10,7 @@ function Game(players, board) {
   this.communityChestCards = new CommunityChestDeck();
   this.shuffleCards();
   this.doublesCount = 0;
+  this.state = 0;
```
Comment (andrewjmeier) [inline src/game.js]: use constant

File: src/player.js [hunk 5]
```diff
@@ -78,4 +78,38 @@ Player.prototype.move = function(spacesToMove) {
...
+
+Player.prototype.buy = function(property) { 
+  this.makePayment(property.cost);
+  this.properties.push(property);
+  property.owner = this;
+};
+
+Player.prototype.assets = function() { 
+
+  var assets = 0;
```
Comment (andrewjmeier) [inline src/player.js]: `var assets = this.money`

**Comment by andrewjmeier** [comment]
Lets get this merged too and try and incorporate new buttons w/ the graphics

File: src/board/board/space.js (hunk 1 above)
Comment (konnorbeard) [inline src/board/board/space.js]: I guess I felt like it was repetitive? don't really remember making this change.

File: src/board/properties/housingProperty.js (hunk 2 above)
Comment (konnorbeard) [inline src/board/properties/housingProperty.js]: just a safety check, but this should never really be called if there's no owner

File: src/game.js (hunk 4 above)
Comment (konnorbeard) [inline src/game.js]: fixed

File: src/player.js (hunk 5 above)
Comment (konnorbeard) [inline src/player.js]: fixed

File: src/board/properties/housingProperty.js (hunk 2 above)
Comment (andrewjmeier) [inline src/board/properties/housingProperty.js]: hm maybe this should be an assert or something b/c we would want to know about that kind of error.

**Comment by andrewjmeier** [comment]
merge conflicts but then :+1:

**Pull request merged**